inspector/src/components/actionTabs/tabs/propertyGrids/animations/addAnimation.tsx

```diff
@@ -26,46 +29,123 @@ export class AddAnimation extends React.Component<
     animationType: string;
     loopMode: number;
     animationTargetPath: string;
+    isUpdating: boolean;
   }
 > {
   constructor(props: IAddAnimationProps) {
     super(props);
+
+    if (this.props.selectedToUpdate !== undefined) {
+      this.setInitialEditingState(this.props.selectedToUpdate);
+    } else {
+      this.setInitialState();
+    }
+  }
+
+  setInitialState() {
     this.state = {
       animationName: '',
       animationTargetPath: '',
       animationType: 'Float',
       loopMode: Animation.ANIMATIONLOOPMODE_CYCLE,
       animationTargetProperty: '',
+      isUpdating: this.props.selectedToUpdate ? true : false,
     };
   }
 
+  setInitialEditingState(editingAnimation: Animation) {
+    this.state = {
+      animationName: editingAnimation.name,
+      animationTargetPath: '',
+      animationType: this.getTypeAsString(editingAnimation.dataType),
+      loopMode: editingAnimation.loopMode ?? Animation.ANIMATIONLOOPMODE_CYCLE,
+      animationTargetProperty: editingAnimation.targetProperty,
+      isUpdating: true,
+    };
+  }
+
+  componentWillReceiveProps(nextProps: IAddAnimationProps) {
+    if (
+      nextProps.selectedToUpdate !== undefined &&
+      nextProps.selectedToUpdate !== this.props.selectedToUpdate
+    ) {
+      this.setInitialEditingState(nextProps.selectedToUpdate);
+    } else {
+      if (nextProps.isOpen === true && nextProps.isOpen !== this.props.isOpen)
+        this.setInitialState();
+    }
+  }
+
+  updateAnimation() {
+    if (this.props.selectedToUpdate !== undefined) {
+      const oldNameValue = this.props.selectedToUpdate.name;
+      this.props.selectedToUpdate.name = this.state.animationName;
+      this.raiseOnPropertyUpdated(
+        oldNameValue,
+        this.state.animationName,
+        'name'
+      );
+
+      const oldLoopModeValue = this.props.selectedToUpdate.loopMode;
+      this.props.selectedToUpdate.loopMode = this.state.loopMode;
+      this.raiseOnPropertyUpdated(
+        oldLoopModeValue,
+        this.state.loopMode,
+        'loopMode'
+      );
+
+      const oldTargetPropertyValue = this.props.selectedToUpdate.targetProperty;
+      this.props.selectedToUpdate.targetProperty = this.state.animationTargetProperty;
+      this.raiseOnPropertyUpdated(
+        oldTargetPropertyValue,
+        this.state.animationTargetProperty,
+        'targetProperty'
+      );
+
+      this.props.finishedUpdate();
+    }
+  }
+
   getAnimationTypeofChange(selected: string) {

```

Comment:
Why aren't we simply using Animation.ANIMATIONTYPE? why do we need the ANIMATIONTYPE-toString-ToAnimationType conversion?

Reply:
I think I'm using the string value in the select input (this forces me to switch between string and int)... but I think I can refactor this to use only integers... let me give it some thought...